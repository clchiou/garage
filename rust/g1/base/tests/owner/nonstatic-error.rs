struct Struct<'a>(&'a [u8]);

impl<'a> TryFrom<&'a [u8]> for Struct<'a> {
    type Error = &'a [u8];

    fn try_from(_: &'a [u8]) -> Result<Self, Self::Error> {
        std::unreachable!()
    }
}

g1_base::define_owner!(Owner for Struct);

fn main() {
    Owner::try_from(Vec::new()).unwrap();
}
