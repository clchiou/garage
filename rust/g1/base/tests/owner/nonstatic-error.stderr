error: higher-ranked lifetime error
  --> tests/owner/nonstatic-error.rs:14:5
   |
14 |     Owner::try_from(Vec::new()).unwrap();
   |     ^^^^^^^^^^^^^^^
